modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/ProtocolType.java

```diff
@@ -90,19 +99,30 @@ public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
 
-        ProtocolType that = (ProtocolType) o;
-
-        if (!protocolName.equals(that.protocolName)) return false;
-        if (!version.equals(that.version)) return false;
+        /* We're using only the pre-calculated hashcode value here since we expect to have ony a limited number
+          * of protocol types in the system and each hash code is calculated using the string values
+          * protocolName and version.
+         */
+        if (this.hashCode == o.hashCode()) {
+            return true;
+        } else {
+            return false;
+        }
+    }
 
-        return true;
+    /**
+     * Since there are limited set of ProtocolType objects available, we can keep the hash code pre calculated
+     * and the the hashcode number set will only grow when a new protocol is plugged in to Andes.
+     */
+    private void generateHashCode() {
+        int result = protocolName.hashCode();
+        result = 31 * result + version.hashCode();
```

Comment:
Since we are using the addition of two integers to create the hash code, to distinguish between a (X + Y) and a (Y + X) we use a constant C such as hashCode = (C*X +Y). In most of the java native classes they have used the number 31 as C, I guess due to the reason mentioned in [1]. This has to do with the fact that in some old CPUs, multiplying by 31 is cheap.

[1] - http://stackoverflow.com/questions/299304/why-does-javas-hashcode-in-string-use-31-as-a-multiplier